flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/NetworkBufferPool.java

```diff
@@ -73,17 +75,31 @@
 
 	private final int numberOfSegmentsToRequest;
 
+	private final long requestSegmentsTimeoutInMillis;
+
+	public NetworkBufferPool(int numberOfSegmentsToAllocate, int segmentSize, int numberOfSegmentsToRequest) {
+		this(numberOfSegmentsToAllocate, segmentSize, numberOfSegmentsToRequest, Integer.MAX_VALUE);
+	}
+
 	/**
 	 * Allocates all {@link MemorySegment} instances managed by this pool.
 	 */
-	public NetworkBufferPool(int numberOfSegmentsToAllocate, int segmentSize, int numberOfSegmentsToRequest) {
+	public NetworkBufferPool(
+		int numberOfSegmentsToAllocate,
+		int segmentSize,
+		int numberOfSegmentsToRequest,
+		long requestSegmentsTimeoutInMillis) {
```

Comment:
I would make this a Duration as well. Always simpler, because it encapsulates the unit, and there is no accidental confusion between millis, nanos, secs, etc.

Reply:
Agree with that an encapsulated object should be more clear and changed the related variables to Duration in `NettyShuffleEnvironmentConfiguration`, `NetworkBufferPool` and `NettyShuffleEnvironmentBuilder`.